appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -107,36 +109,59 @@ internal class DebuggerViewModel(
         }
     }
 
-    fun onStart(deepLinkPath: String?) {
-        if (deepLinkPath.isNullOrEmpty()) {
-            // if no path is given, and currently expanded, go back to idle
-            if (_uiState.value is Expanded) {
-                _uiState.value = Idle
+    fun onStart(mode: DebugMode, reset: Boolean) {
+        this.mode = mode
+
+        when (mode) {
+            is Debugger -> {
+                val deepLinkPath = mode.deepLinkPath
+                if (deepLinkPath.isNullOrEmpty()) {
+                    // if no path is given, and currently expanded, go back to Idle
+                    if (_uiState.value is Expanded ||
+                        // OR if showing FAB but the mode changed, reset to new Idle mode
+                        (_uiState.value is Idle && reset)) {
+                        _uiState.value = Idle(mode)
+                    }
+                    return
+                } else {
+                    // the deep link might be from the debugger checking that the configuration is correct
+                    // pass along to allow it to check
+                    viewModelScope.launch {
+                        debuggerStatusManager.checkDeepLinkValidation(deepLinkPath)
+                    }
+                }
+
+                when (_uiState.value) {
+                    // if currently idle and a new link comes in - expand to that link
+                    is Idle -> _uiState.value = Expanded(mode)
+                    // currently open - reset to the new path
+                    is Expanded -> _uiState.value = Expanded(mode)
+                    // otherwise, no valid link action available
+                    else -> Unit
+                }
             }
-            return
-        } else {
-            // the deep link might be from the debugger checking that the configuration is correct
-            // pass along to allow it to check
-            viewModelScope.launch {
-                debuggerStatusManager.checkDeepLinkValidation(deepLinkPath)
+            is ScreenCapture -> {
+                // if in debugger mode, expanded, go to new Idle state
+                if (_uiState.value is Expanded ||
+                    // OR if in Idle (FAB) mode but mode changed, reset to new Idle mode
+                    (_uiState.value is Idle && reset)) {
+                    _uiState.value = Idle(mode)
```

Comment:
and here as well.

I could have made an entirely new state for `ScreenCapture` for the FAB, but it seemed like that would have a lot more cascading changes and this would keep it more isolated.